include/exiv2/types.hpp

```diff
@@ -179,7 +179,9 @@ namespace Exiv2 {
      */
     struct EXIV2API DataBufRef {
         //! Constructor
-        explicit DataBufRef(std::pair<byte*, long> rhs) : p(rhs) {}
+        explicit DataBufRef(std::pair<byte*, long> rhs) : p(std::move(rhs))
```

Comment:
I might be having a slow brain day, but why would this be better? pointer to `byte` and `long` aren't movable types, so this isn't "moving" anything

Reply:
Slow indeed. This is an std::pair that holds a pointer. The actual details of std::pair in libc++ are actually pretty complicated. This is here because std::pair is not trivially copyable. Strange I know.

Reply:
Yeah.

```
include/exiv2/types.hpp:185:72: error: static assertion failed
  185 |     static_assert(std::is_trivially_copyable<std::pair<byte*, long>>::value);
```

Reply:
Replacing long with const long makes it trivially copyable....

Reply:
I still don't see how the change would improve the code.  
Can you come up with an example on godbolt.org that shows a difference?   

As far as I understand it, I don't see how this would result in better code generation.  
You are calling the copy or move constructor of `std::pair<byte*, long>` which will both simply copy the `byte*` and `long`.  
There would be a difference if you had types that are movable inside the `std::pair` but not in this case.

Reply:
I'm not s libc++ expert but the warning gets thrown for a reason probably. An std::pair and a struct with two members are not the same. Except in constexpr context.

Reply:
I don't quite follow your argument about trivially copyable and the difference of a pair to a struct, as in I'm not sure why they are relevant to this problem.   

I'm all for understanding what a tool changes, instead of just accepting it as gospel. After all, we are all humans, even the authors of clang-tidy, and we all make mistakes ;)      
Thus, unless we come up with a good reason to change this line as proposed, I'd vote to close the PR.